### PR TITLE
P#72885 Improve configuration of whether reference detail pages should be shown

### PR DESCRIPTION
**Pull request by dai-eastgate**

### CODE REVIEW

**Review by LongTrong-exe — DISMISSED** [review]

**Review by LongTrong-exe — APPROVED** [review]

**Review by jmaas-onoffice — CHANGES_REQUESTED** [review]
Minor nitpick: In the description "... because the access is [restricted](https://testinstanz-jmaa.onofficeweb.com/wp-admin/admin.php?page=onoffice-estates&tab=detail)", there is a period missing at the end. (It should be "... because the access is [restricted](https://testinstanz-jmaa.onofficeweb.com/wp-admin/admin.php?page=onoffice-estates&tab=detail).")

Other than that a quick test of the functionality looks good already! :)

File: plugin/Installer/DatabaseChanges.php [hunk 1]
```diff
@@ -241,12 +241,14 @@ public function install()
 			$dbversion = 31;
 		}
 
-		if ($dbversion == 31) {
+		if ( $dbversion == 31 ) {
 			$this->_pWpOption->addOption('onoffice-is-encryptcredent', false);
+			$this->updateShowReferenceEstate();
+			$this->setDataDetailViewRestrictAccessControlValue();
 			$dbversion = 32;
```
[the remark this answers is not in the document]
Comment (dai-eastgate) [reply plugin/Installer/DatabaseChanges.php:248]: yes, it is. I implement this to migrate the old data.

[the remark this answers is not in the document]
Comment (jmaas-onoffice) [reply plugin/Installer/DatabaseChanges.php:248]: But should this be done when `$dbVersion == 31` or should it get a new version?

[the remark this answers is not in the document]
Comment (dai-eastgate) [reply plugin/Installer/DatabaseChanges.php:248]: I think `$dbVersion == 31` is the current version (that we are developing). Please correct it if I'm wrong. Is the code for `$this->_pWpOption->addOption('onoffice-is-encryptcredent', false);` already release?

[the remark this answers is not in the document]
Comment (jmaas-onoffice) [reply plugin/Installer/DatabaseChanges.php:248]: If `$dbVersion == 31` is in master, do we need a new version? Could you check this?

[the remark this answers is not in the document]
Comment (dai-eastgate) [reply plugin/Installer/DatabaseChanges.php]: I just checked it. We need a new version. I updated the code. Please check it.

[the remark this answers is not in the document]
Comment (jmaas-onoffice) [reply plugin/Installer/DatabaseChanges.php]: Thanks! I'm not an expert on how the code works, so I'm relying on you... :D But I will test it. 👍

**Review by jmaas-onoffice — APPROVED** [review]

**Comment by jmaas-onoffice** [comment]
This works great, thanks! I'll just wait and see if we can make the tests pass, by restarting them later.

**Review by jmaas-onoffice — APPROVED** [review]

**Review by jmaas-onoffice — CHANGES_REQUESTED** [review]
Unfortunately, I forgot to test a few cases and I found a migration problem:

If on v4.2 in the detail view I have set the reference estates to be accessible:

grafik

And then I upgrade, suddenly the reference pages are restricted:

grafik

This also happens if on v4.2 the reference detail pages are not accessible. That means that with the update, the reference pages are always set to be restricted.

But for existing installations, the reference estates detail page should keep the accessibility from before the update. Only for new installations should the default be that they are restricted.

**Review by jmaas-onoffice — APPROVED** [review]

**Comment by jmaas-onoffice** [comment]
When the problem with the unit tests is solved, we can merge this.

**Pull request merged**